config/initializers/gds-api-adapters.rb

```diff
@@ -0,0 +1 @@
+GdsApi.config.always_raise_for_not_found = true
```

Comment:
Actually, doesn't look like this is too onerous as it seems all the API interactions already uses hash accessors.  I'll do this too.